tools/generator/Java.Interop.Tools.Generator.ObjectModel/ClassGen.cs

```diff
@@ -334,21 +232,94 @@ public static void GenerateTypeRegistrations (CodeGenerationOptions opt, Generat
 			}
 		}
 
-		public static void GenerateEnumList (GenerationInfo gen_info)
-		{
-			using (var sw = new StreamWriter (File.Create (Path.Combine (gen_info.CSharpDir, "enumlist")))) {
-				foreach (string e in gen_info.Enums.OrderBy (p => p, StringComparer.OrdinalIgnoreCase))
-					sw.WriteLine (e);
-			}
-		}
-		
 		protected override bool GetEnumMappedMemberInfo ()
 		{
 			foreach (var m in Ctors)
 				return true;
+
 			return base.GetEnumMappedMemberInfo ();
 		}
-		
+
+		internal IEnumerable<InterfaceExtensionInfo> GetNestedInterfaceTypes ()
+		{
+			var nestedInterfaces = new List<InterfaceExtensionInfo> ();
+			AddNestedInterfaceTypes (this, nestedInterfaces);
+			return nestedInterfaces;
+		}
+
+		public bool InheritsObject { get; set; }
+
+		public bool IsAbstract { get; set; }
+
+		public bool IsExplicitlyImplementedMethod (string sig)
+		{
+			for (var c = this; c != null; c = c.BaseGen)
+				if (c.ExplicitlyImplementedInterfaceMethods.Contains (sig))
+					return true;
+			return false;
+		}
+
+		public bool IsFinal { get; set; }
+
+		public bool NeedsNew { get; set; }
+
+		protected override bool OnValidate (CodeGenerationOptions opt, GenericParameterDefinitionList type_params, CodeGeneratorContext context)
```

Comment:
Why "re-order" this method?  It appears to be nearly identical to the `OnValidate()` method which was "removed" earlier in the patch.

Ditto `ToNative()`, `FromNative()`, `ResetValidation()`.